src/toil/wdl/wdl_functions.py

```diff
@@ -511,26 +512,28 @@ def size(f: Optional[Union[str, WDLFile, List[Union[str, WDLFile]]]] = None,
     assert isinstance(f, (WDLFile, list)), f'size() excepts a "File" or "File?" argument!  Not: {type(f)}'
 
     # validate the input. fileStore is only required if the input is not processed.
-    f = process_infile(f, fileStore)
+    if fileStore:
+        f = process_infile(f, fileStore)
 
     divisor = bytes_in_unit(unit)
 
     if isinstance(f, list):
-        total_size = sum(file.file_path.size for file in f)
+        total_size = sum(file.file_path.size for file in f if isinstance(file, WDLFile))
```

Comment:
Aren't they all WDLFiles by now because f was replaced above in `elif isinstance(f, list):`?

Reply:
They would all be WDLFiles there but mypy was throwing the error `src/toil/wdl/wdl_functions.py:521: error: Item "str" of "Union[str, WDLFile]" has no attribute "file_path"` since it isn't able to infer that. I don't think we can prevent the error without specifying they are WDLFiles again so maybe we could include `# type: ignore` here.

Reply:
The way to fix this is to do less variable overwriting. Switching or consolidating types? Make a new variable 😎